External/Traffic/Source/MassTraffic/Private/MassTrafficVehicleVisualizationProcessor.cpp

```diff
@@ -242,12 +244,27 @@ void UMassTrafficVehicleUpdateCustomVisualizationProcessor::Execute(FMassEntityM
 			const FMassEntityHandle Entity = Context.GetEntity(EntityIdx);
 
 			const FMassTrafficRandomFractionFragment& RandomFractionFragment = RandomFractionFragments[EntityIdx];
-			const FMassTrafficVehicleLightsFragment& VehicleStateFragment = VehicleStateFragments[EntityIdx];
 			const FTransformFragment& TransformFragment = TransformFragments[EntityIdx];
 			const FMassRepresentationLODFragment& RepresentationLODFragment = RepresentationLODFragments[EntityIdx];
+			FMassTrafficVehicleLightsFragment& VehicleStateFragment = VehicleStateFragments[EntityIdx];
 			FMassActorFragment& ActorFragment = ActorFragments[EntityIdx];
 			FMassRepresentationFragment& RepresentationFragment = VisualizationFragments[EntityIdx];
 
+			if (!EnvironmentalBrightnessFragments.IsEmpty())
+			{
+				const FEnvironmentalBrightnessFragment& EnvironmentalBrightnessFragment = EnvironmentalBrightnessFragments[EntityIdx];
+
+				// Automatically turn headlights on and off with the vehicle's current level of environmental brightness.
+				VehicleStateFragment.bHeadlights = (!VehicleStateFragment.bHeadlights && EnvironmentalBrightnessFragment.Brightness <= MassTrafficSettings->VehicleTurnOnHeadlightsBrightnessThreshold) ||
```

Comment:
Is the "or equal to" important here? (maybe it was before you separated the turn on and off thresholds?) If not, let's remove it?

Reply:
I always had the 2 thresholds.  But, the "or equal to" isn't really important.  It's just meant to give symmetry to the `off -> on` and `on -> off` transitions.

The `on -> off` transition happens when `EnvironmentalBrightnessFragment.Brightness < MassTrafficSettings->VehicleTurnOffHeadlightsBrightnessThreshold` is false.  Or, in other words via an "implied" `EnvironmentalBrightnessFragment.Brightness >= MassTrafficSettings->VehicleTurnOffHeadlightsBrightnessThreshold`, which includes the threshold value itself.